src/runtime_src/core/common/device.h

```diff
@@ -265,7 +265,7 @@ class device : public ishim
    */
   virtual void write(uint64_t, const void*, uint64_t) const {}
 
-  virtual void reset(query::reset_type) const {}
+  virtual void reset(query::reset_type&) const {}
```

Comment:
Is this function modifying the argument?  If not please use `const`
I see caller does not modify it, but that the get methods of query_reset are not declared const as they should be.  Please change this also.

Reply:
Seems like the arg is getting modified. This code is coming from #3939 and Stephen will be working on fixing this.